android/app/src/main/java/com/microsoft/codepush/react/CodePushUpdateManager.java

```diff
@@ -345,35 +364,49 @@ public void rollbackPackage() {
     }
 
     public void downloadAndReplaceCurrentBundle(String remoteBundleUrl, String bundleFileName) throws IOException {
-        URL downloadUrl;
-        HttpURLConnection connection = null;
-        BufferedInputStream bin = null;
-        FileOutputStream fos = null;
-        BufferedOutputStream bout = null;
         try {
-            downloadUrl = new URL(remoteBundleUrl);
-            connection = (HttpURLConnection) (downloadUrl.openConnection());
-            bin = new BufferedInputStream(connection.getInputStream());
-            File downloadFile = new File(getCurrentPackageBundlePath(bundleFileName));
-            downloadFile.delete();
-            fos = new FileOutputStream(downloadFile);
-            bout = new BufferedOutputStream(fos, CodePushConstants.DOWNLOAD_BUFFER_SIZE);
-            byte[] data = new byte[CodePushConstants.DOWNLOAD_BUFFER_SIZE];
-            int numBytesRead = 0;
-            while ((numBytesRead = bin.read(data, 0, CodePushConstants.DOWNLOAD_BUFFER_SIZE)) >= 0) {
-                bout.write(data, 0, numBytesRead);
-            }
+            final URL downloadUrl = new URL(remoteBundleUrl);
+            final String finalBundleFileName = bundleFileName;
+            
+            RetryHelper.executeWithRetry(new RetryHelper.RetryableOperation() {
+                @Override
+                public void execute() throws IOException {
+                    HttpURLConnection connection = null;
+                    BufferedInputStream bin = null;
+                    FileOutputStream fos = null;
+                    BufferedOutputStream bout = null;
+                    
+                    try {
+                        connection = (HttpURLConnection) (downloadUrl.openConnection());
+                        connection.setConnectTimeout(30000);
+                        connection.setReadTimeout(60000);
```

Comment:
Is there any reason to set timeout as 30s and 60s? It seems to be too long for me 🤔